### PR TITLE
KAADO-30: Implement card share service

### PR DESCRIPTION
**Pull request by nasudon**
- Implement card share service

### CODE REVIEW

File: src/services/shareService.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        const { types } = await getTypes();
+        if (!types) {
+            return { error: 'Unable to copy card' };
+        }
+        const defaultType = types.filter(t => t.isDefault)[0];
+        if (!defaultType) {
+            return { error: 'Unable to copy card' };
+        }
+        const newCard = await createCard(new CardAdaptor({image_front: card.image_front, image_back: card.image_back, name: card.name, type_id: defaultType._id!, number: card.number, note: card.note}));
+        if (newCard) {
```
Comment (tsoibet) [inline src/services/shareService.ts:152]: createCard returns an object of { card } or { error }.

File: src/services/shareService.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function createSharing(cardId: string, validHours = (24*7)) {
+    try {
+        await connectDB();
+        
+        const card = stringToObjectId(cardId);
```
Comment (tsoibet) [inline src/services/shareService.ts]: I think the variable `card` is objectId instead of Card object.

**Review by tsoibet — COMMENTED** [review]
Please see conversations

File: src/services/shareService.ts (hunk 1 above)
Comment (nasudon) [reply src/services/shareService.ts:152]: Fixed: https://github.com/tsoibet/kaado/pull/14/commits/ab2246d599ef556371ce94663f0958bd0ee15837

File: src/services/shareService.ts (hunk 2 above)
Comment (nasudon) [reply src/services/shareService.ts]: Fixed: https://github.com/tsoibet/kaado/pull/14/commits/425ff6b2bd5601e42d11f7ef8697111da4cd020f

**Review by tsoibet — APPROVED** [review]
LGTM

**Pull request merged**